src/Product/publish.ts

```diff
@@ -0,0 +1,49 @@
+import assert from 'assert';
+import { HttpModule } from '@whppt/api-express';
+import { Product } from './Models/Product';
+
+const changeDetails: HttpModule<{ productId: string }, void> = {
+  authorise({ $identity }, { user }) {
+    return $identity.isUser(user);
+  },
+  exec({ $salesforce, $database, createEvent }, { productId }) {
+    assert(productId, 'Product Id is required.');
+
+    return $database
+      .then(({ document, startTransaction }) => {
+        return document.fetch<Product>('products', productId).then(product => {
+          assert(product, 'Product does not exsist');
+          const event = createEvent('ProductPublished', product);
+
+          return startTransaction(session => {
+            return Promise.all([
+              document.saveWithEvents('products', product, [event], { session }),
+              document.publish('products', product, { session }),
+            ]).then(() => {
+              return $salesforce.$Oauth().then((token: string) => {
```

Comment:
This ok for now but we'll need a disconnected solution. Can we put a TODO in to remind us